x-pack/plugins/security_solution/server/lib/telemetry/async_sender.ts

```diff
@@ -0,0 +1,397 @@
+        )
+      ),
+
+      // update inflight events counter
+      rx.tap((result: Result) => {
+        inflightEvents$.next(-result.events);
+      })
+    ) as rx.Observable<Result>;
+  }
+
+  private async sendEvents(channel: TelemetryChannel, events: string[]): Promise<Result> {
+    try {
+      const senderMetadata = await this.getSenderMetadata(channel);
+
+      const isTelemetryOptedIn = await senderMetadata.isTelemetryOptedIn();
+      if (!isTelemetryOptedIn) {
+        this.senderUtils?.incrementCounter(
+          TelemetryCounter.TELEMETRY_OPTED_OUT,
+          events.length,
+          channel
+        );
+
+        this.logger.l('Telemetry is not opted-in.');
+        return Promise.reject(newFailure('Telemetry is not opted-in', channel, events.length));
+      }
+
+      const isElasticTelemetryReachable = await senderMetadata.isTelemetryServicesReachable();
+      if (!isElasticTelemetryReachable) {
+        this.logger.l('Telemetry Services are not reachable.');
+        this.senderUtils?.incrementCounter(
+          TelemetryCounter.TELEMETRY_NOT_REACHABLE,
+          events.length,
+          channel
+        );
+
+        return Promise.reject(
+          newFailure('Telemetry Services are not reachable', channel, events.length)
+        );
+      }
+
+      this.logger.l(`Sending ${events.length} telemetry events to ${channel}`);
+
+      const body = events.join('\n');
+
+      const telemetryUrl = senderMetadata.telemetryUrl;
+
+      // TODO(sebastian.zaffarano): When send events to `alerts-endpoint` using
+      // `sender.queueEvents`, it adds cluster and license information to the event:
+      // ...(licenseInfo ? { license: this.receiver?.copyLicenseFields(licenseInfo) } : {}),
+      // cluster_uuid: clusterInfo?.cluster_uuid,
+      // cluster_name: clusterInfo?.cluster_name,
+      // do we need to do it? diagnostic.ts also sends to the same channel, but
+      // using `sendOnDemand` and hence it doesn't enrich the original-body
+
+      return await axios
+        .post(telemetryUrl, body, {
+          headers: senderMetadata.telemetryRequestHeaders(),
+          timeout: 10000,
+        })
+        .then((r) => {
+          this.senderUtils?.incrementCounter(
+            TelemetryCounter.HTTP_STATUS,
+            events.length,
+            channel,
+            r.status.toString()
+          );
+
+          if (r.status < 400) {
+            return { events: events.length, channel };
+          } else {
+            this.logger.l(`Unexpected response, got ${r.status}`);
+            throw newFailure(`Got ${r.status}`, channel, events.length);
+          }
+        })
+        .catch((err) => {
+          this.senderUtils?.incrementCounter(
+            TelemetryCounter.RUNTIME_ERROR,
+            events.length,
+            channel
+          );
+
+          this.logger.l(`Runtime error: ${err.message}`);
+          throw newFailure(`Error posting events: ${err}`, channel, events.length);
```

Comment:
Wouldn't this error be caught on `L329` and then this error would be counted for both `RUNTIME_ERROR` and `FATAL_ERROR`. Was that intended?

Reply:
yes! good catch!
I've just pushed a fix to avoid it